### ISSUE TITLE
Why is this code related to the arxiv paper OSNet

### ISSUE THREAD

**Issue by ba305** [KaiyangZhou/deep-person-reid#182]
Hi, sorry if I am mistaken, but this GitHub repo does not seem to match the Arxiv paper cited in the README (Omni-Scale feature learning, or OSNet). Is that correct? Thanks

**Comment by KaiyangZhou** [comment]
Good question. Let me clarify this.

**This code was originally developed for OSNet**. Unlike most papers, we released the code far before the release of this paper so that it could be beneficial to the reid community.

As you can see OSNet is an architecture paper, which required extensive efforts in researching existing CNN architectures both for image classification (e.g. ResNet, MobileNet) and reid (e.g. MLFN, HACNN), as well as identifying what are useful to reid and what are not. Therefore, you could see that this repo contains implementations of those architectures. With this code, you can easily test your models on different datasets using the standard protocols which align with the literature.

More importantly, this code is the first one to disclose some advanced training techniques for reid such as label smoothing (which was not even mentioned in papers), by which the performance can be significantly boosted. As mentioned in readme, the code of OSNet will be released.

Therefore, I would suggest if this code is useful to your research, please give us some credit.

**Comment by ba305** [comment]
I see, that is very helpful, thank you for your response. We definitely appreciate that you decided to release Torchreid to the community.

So as of now, if I use the Triplet engine, is it fair to say that this is essentially the architecture explained in "In Defense of the Triplet Loss" (with a few modifications, such as label smoothing)? Or is there another major difference?

Also am really looking forward to seeing code for OSNet, it sounds like an important contribution to the current state-of-the art.

Thank you!

**Comment by KaiyangZhou** [comment]
for the triplet, if you set `weight_t=1, weight_x=0` (disable softmax loss), it is identical to the "In defence" paper, pls see `scripts/default_parser.py` regarding the arguments

**Comment by ba305** [comment]
Makes sense. Thanks for your help!
